docs/development/local_setup.md

```diff
@@ -68,9 +70,67 @@ Currently there are no binary builds available, but it is fairly simple to build
 You can verify the installation by running following command:
 
     ```console
-    $ etcdbrctl -v
-    INFO[0000] etcd-backup-restore Version: v0.7.0-dev
-    INFO[0000] Git SHA: 38979f0
-    INFO[0000] Go Version: go1.12
-    INFO[0000] Go OS/Arch: darwin/amd64
+    ➜ etcdbrctl -v
+    INFO[0000] etcd-backup-restore Version: v0.29.0-dev     
+    INFO[0000] Git SHA: 0247d8d4                            
+    INFO[0000] Go Version: go1.22.1                         
+    INFO[0000] Go OS/Arch: darwin/arm64
     ```
+
+## Running `etcdbrctl` as a process
+
+> [!WARNING]  
+> Ensure that you do not add your `credentialDirectory` or `credential_file.json` to your version control. Easiest way to ensure this, is to avoid using `git add .`. You could add these files/directory as `.gitignore` targets, or have the credential file/directory exist in a directory which is not tracked by `git`.
+
+### Environment variables
+
+Taking the example of S3 here, one has to:
+
+* Set the `AWS_APPLICATION_CREDENTIALS` environment variable if the credentials are to be stored in a directory, where each file in the directory holds a different credential
+* Set the `AWS_APPLICATION_CREDENTIALS_JSON` if the credentials are to be stored in `.json` file.
```

Comment:
I think it is better to also include the sole environment variables way of setting the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_REGION`, and `AWS_ENDPOINT_URL_S3` ( if the user wants to test using an instance of localstack emulator ) environment variables.

Maybe we can say that this should be the preferred way as otherwise, there is always a chance of exposing the credentials in public if we are using files to authenticate.

Reply:
In this specific case, I only wanted to give an example of how the credentials must look, by taking the example of S3. Usage of `etcdbrctl` with a cloud provider is out of scope for this section, in my opinion.

I don't want to mix up documentation of running `etcdbrctl` as a process with using `etcdbrctl` with emulators. I think there could be a separate section which explains how to use `etcdbrctl` with all three emulators? In this section we can describe the environment variable method of passing credentials as you mentioned.

What would you suggest?

Reply:
Yeah I'm okay with not including emulators in this doc. My main point was to use environment variables for the creds, which you are okay with. All good.

Reply:
Passing credentials purely through environment variables `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_REGION` is currently not supported in `etcd-backup-restore`. For that reason, @anveshreddy18 and I decided that it wouldn't be necessary to add this to the current documentation.